packages/monaco-editor-wrapper/src/editorAppBase.ts

```diff
@@ -223,3 +227,56 @@ export abstract class EditorAppBase {
+    let updateRequired = modelUpdate.code !== undefined && modelUpdate.code !== config.code;
+    updateRequired = updateRequired || modelUpdate.codeOriginal !== config.codeOriginal;
+    return updateRequired ? ModelUpdateType.code : ModelUpdateType.none;
+};
+
+export const isModelUpdateRequired = (config: EditorAppBaseConfig, modelUpdate: ModelUpdate): ModelUpdateType => {
+    const codeUpdate = isCodeUpdateRequired(config, modelUpdate);
+    let updateRequired = modelUpdate.languageId !== config.languageId;
+    updateRequired = updateRequired || modelUpdate.codeUri !== config.codeUri;
+    updateRequired = updateRequired || modelUpdate.codeOriginalUri !== config.codeOriginalUri;
+    return updateRequired ? ModelUpdateType.model : codeUpdate;
+};
+
+export enum ModelUpdateType {
+    none,
+    code,
+    model
+}
+
+export const isAppConfigDifferent = (orgConfig: EditorAppConfigClassic | EditorAppConfigVscodeApi,
+    config: EditorAppConfigClassic | EditorAppConfigVscodeApi, includeModelData: boolean, includeEditorOptions: boolean): boolean => {
+
+    // this is done by hand, ModelUpdate is only considered if flag is set
+    let different = includeModelData ? isModelUpdateRequired(orgConfig, config) !== ModelUpdateType.none : false;
+    if (orgConfig.$type === config.$type) {
+        if (orgConfig.$type === 'classic' && config.$type === 'classic') {
+            different = different || orgConfig.automaticLayout !== config.automaticLayout;
+            different = different || orgConfig.domReadOnly !== config.domReadOnly;
+            if (includeEditorOptions === true) {
+                different = different || orgConfig.diffEditorOptions !== config.diffEditorOptions;
+                different = different || orgConfig.editorOptions !== config.editorOptions;
+            }
+            different = different || orgConfig.languageDef !== config.languageDef;
+            different = different || orgConfig.languageExtensionConfig !== config.languageExtensionConfig;
+            different = different || orgConfig.readOnly !== config.readOnly;
+            different = different || orgConfig.theme !== config.theme;
+            different = different || orgConfig.themeData !== config.themeData;
+            different = different || orgConfig.useDiffEditor !== config.useDiffEditor;
+        } else if (orgConfig.$type === 'vscodeApi' && config.$type === 'vscodeApi') {
+            different = different || orgConfig.domReadOnly !== config.domReadOnly;
+            different = different || orgConfig.extension !== config.extension;
+            different = different || orgConfig.extensionFilesOrContents !== config.extensionFilesOrContents;
+            different = different || orgConfig.readOnly !== config.readOnly;
+            different = different || orgConfig.useDiffEditor !== config.useDiffEditor;
+            different = different || orgConfig.userConfiguration !== config.userConfiguration;
+        }
```

Comment:
I would rewrite this as either `[predicate1,predicate2,...].some(p => p)` or a series of branches. With so many repeated assignments it seems like it would be hard to find a small mistake in the logic. Given if a difference is detected when any case is satisfied, returning a conditional directly would allow short-circuiting to take effect. This would save the repeated assignments, and leave the code for each case as just that case, instead of an assignment + the case.

Reply:
We discussed offline that no further generalization is needed.